VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.ts

```diff
@@ -30,6 +31,17 @@ export class SelectHearingLayoutComponent implements OnInit {
         this.setAccordionText({} as MouseEvent);
     }
 
+    ngOnChanges(): void {
```

Comment:
Should this not be in ngOnInit and then unsubscribed in ngOnDestroy. Won't subscribing here set up multiple listeners?

Reply:
Do you mean to move the onlangchange event into the ngoninit function? I didn't test that way but wouldn't that not update the page without a hard reload?

Reply:
Yeah, no it shouldn't as you are subscribing to an event and then just updating some values.

Reply:
removed the lifecycle method, I thought it gets called whenever the state changes in the app, but then I realized we actually dont have a global state instead we use local/session storage.